src/components/SelectedAnnotation.jsx

```diff
@@ -323,11 +320,22 @@ class SelectedAnnotation extends React.Component {
     this.props.onClose();  //Note that deleteSelectedAnnotation() also runs unselectAnnotation(), but this needs to be called anyway to inform the parent component.
   }
 
+  removeEventListeners() {
+    this.inputText.removeEventListener('mouseup', () => {
```

Comment:
Wait, I just realised this doesn't work whatsoever - even our current code is wrong.

When calling...
```
htmlElement.addEventListener(eventName, eventHandler);
htmlElement.removeEventListener(eventName, eventHandler);
```
... `eventHandler` has to be the _same thing._

So you can have, say...
```
const eventName = 'mouseup';
const eventHandler = this.handleMouseUp;

htmlElement.addEventListener(eventName, eventHandler);
htmlElement.removeEventListener(eventName, eventHandler);
```
...and it'll work fine.

However, calling...
```
htmlElement.addEventListener(eventName, () => { console.log('This is anonymous handler A') });
htmlElement.removeEventListener(eventName, () => { console.log('This is anonymous handler B') });
```
...means that we _keep adding_ (anonymous) event listeners to `htmlElement` and never remove any.

I suspect this hasn't been an issue for us because...
- when the React component unmounts, it annihilates the `htmlElement` itself, so it doesn't matter if it was overloaded with six hundred anonymous event handlers.
- in the case of `document.addEventListener`, we properly use a common reference event handler (`this.handleKeyUp`)

Recommendations:
- Create `SelectedAnnotation.handleKeyUp` and `SelectedAnnotation.handleKeyDown` methods.
- Make sure all `this.whatever.removeEventListener()` calls reference those methods instead of generating new anonymous ones.

Reply:
React destroys any event listeners that it created from component props, on unmount. If I remember the React docs correctly, it doesn't remove any listeners that you added with `addEventListener`.

[React says it supports `onMouseUp` now](https://reactjs.org/docs/events.html#mouse-events), so you might not have to manually add these listeners yourselves.